app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -1717,6 +1733,10 @@ private void onUpdateProgress(final int currentProgress,
         }
         binding.playbackLiveSync.setClickable(!isLiveEdge());
 
+        if (isLiveEdge() && getPlaybackSpeed() > 1.0f) {
```

Comment:
Did you notice that this is run constantly in a loop?

Reply:
I did realize this is a function that is executing very frequently, but I wanted the playback speed to reset the moment the live edge was reached and wasn't sure if there was a better place to put the logic. 

This logic could be moved to `onBuffering()`, which would match Youtube's livestreaming behavior and avoid calling the check in a loop. The downside is that the speed reset won't occur immediately after hitting the live edge.

Reply:
The more resource-heavy part here is probably [calculating `isLiveEdge`](https://github.com/TeamNewPipe/NewPipe/blob/81dd93e8bfd5dea32b777af03a42577094f24c9c/app/src/main/java/org/schabi/newpipe/player/Player.java#L2768), seeing it is not a trivial function.